lib/modules/datasource/github-runners/index.ts

```diff
@@ -0,0 +1,39 @@
+import { id as dockerVersioningId } from '../../versioning/docker';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+export class GithubRunnersDatasource extends Datasource {
+  static readonly id = 'github-runners';
+
+  /**
+   * Only add stable runners to the datasource. See datasource readme for details.
+   */
+  private static readonly releases: { [packageName: string]: Release[] } = {
+    ubuntu: [{ version: '22.04' }, { version: '20.04' }],
+    macos: [
+      { version: '13' },
+      { version: '13-xl' },
+      { version: '12' },
+      { version: '12-xl' },
+      { version: '11' },
+    ],
+    windows: [{ version: '2022' }, { version: '2019' }],
+  };
+
+  override readonly defaultVersioning = dockerVersioningId;
+
+  constructor() {
+    super(GithubRunnersDatasource.id);
+  }
+
+  override getReleases({
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult> {
+    const releaseResult: ReleaseResult = {
+      releases: GithubRunnersDatasource.releases[packageName] || [],
```

Comment:
```suggestion
      releases: GithubRunnersDatasource.releases[packageName] ?? [],
```
the else path is probably not taken in tests

https://app.codecov.io/gh/renovatebot/renovate/pull/23633/blob/lib/modules/datasource/github-runners/index.ts#L33

this API should return null instead of empty results.